proto/lbm/fbridge/v1/tx.proto

```diff
@@ -0,0 +1,152 @@
+  rpc UpdateRole(MsgUpdateRole) returns (MsgUpdateRoleResponse);
+
+  // Halt the bridge module.
+  rpc Halt(MsgHalt) returns (MsgHaltResponse);
+
+  // Resume the bridge module.
+  rpc Resume(MsgResume) returns (MsgResumeResponse);
+}
+
+// MsgTransfer is input values required for bridge transfer
+message MsgTransfer {
+  // the sender address on the source chain
+  string sender = 1;
+  // the recipient address on the destination chain
+  string receiver = 2;
+  // the amount of token to be transferred
+  string amount = 3
+      [(gogoproto.customtype) = "github.com/Finschia/finschia-sdk/types.Int", (gogoproto.nullable) = false];
+}
+
+message MsgTransferResponse {}
+
+// MsgProvision is input values required for provisioning
+message MsgProvision {
+  // the operator address
+  string from = 1;
+  // the sequence number of the bridge request
+  uint64 seq = 2;
+  // the sender address on the source chain
+  string sender = 3;
+  // the recipient address on the destination chain
+  string receiver = 4;
+  // the amount of token to be claimed
+  string amount = 5
+      [(gogoproto.customtype) = "github.com/Finschia/finschia-sdk/types.Int", (gogoproto.nullable) = false];
+}
+
+message MsgProvisionResponse {}
+
+// MsgHoldTransfer is input values required for holding transfer
+message MsgHoldTransfer {
+  // the judge address
+  string from = 1;
+  // the sequence number of the bridge request
+  uint64 seq = 2;
+}
+
+message MsgHoldTransferResponse {}
+
+// MsgReleaseTransfer is input values required for releasing a held transfer by time lock
+message MsgReleaseTransfer {
+  // the guardian address
+  string from = 1;
+  // the sequence number of the bridge request
+  uint64 seq = 2;
+}
+
+message MsgReleaseTransferResponse {}
+
+// MsgRemoveProvision is input values required for removing a specific confirmed provision
+message MsgRemoveProvision {
+  // the judge address
+  string from = 1;
+  // the sequence number of the bridge request
+  uint64 seq = 2;
+}
+
+message MsgRemoveProvisionResponse {}
+
+// MsgClaimBatch is input values required for claiming multiple claimable provisions
+message MsgClaimBatch {
+  // the claimer address
+  string from = 1;
+  // the maximum number of claims to be made at once
+  uint64 max_claims = 2;
+}
+
+message MsgClaimBatchResponse {}
+
+// MsgClaim is input values required for claiming a provision
+message MsgClaim {
+  // the claimer address
+  string from = 1;
+  // the sequence number of the bridge request
+  uint64 seq = 2;
+}
+
+message MsgClaimResponse {}
+
+// MsgUpdateRole is input values required for updating the role of an address
+message MsgUpdateRole {
+  // the guardian address
+  string from = 1;
+  // the address to update the role
+  string target = 2;
+  // the role to be updated
+  // - unspecified : 0, used to remove the address from a group
+  // - guardian : 1
+  // - operator : 2
+  // - judge : 3
+  Role role = 3;
+}
```

Comment:
If there is no `seq` field in the `MsgUpdateRole`, I think we cannot distinguish it from the previous proposal when another proposal with the same param is added. 

So how about adding `seq` field or modifying it with another solution?

Reply:
I separated voting from submission like you suggested before. So the proposal ID(`seq`) will be assigned automatically.